src/analysis/retail/evoker/devastation/modules/abilities/Disintegrate.tsx

```diff
@@ -0,0 +1,61 @@
+import SPELLS from 'common/SPELLS';
+import { TALENTS_EVOKER } from 'common/TALENTS';
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Events from 'parser/core/Events';
+
+const { DISINTEGRATE } = SPELLS;
+
+const { DRAGONRAGE_TALENT } = TALENTS_EVOKER;
+
+const TICKS_PER_DISINTEGRATE = 4;
+
+class Disintegrate extends Analyzer {
+  totalCasts: number = 0;
+  totalTicks: number = 0;
+  dragonRageTicks: number = 0;
+  dragonRageCasts: number = 0;
+  inDragonRageWindow: boolean = false;
+
+  constructor(options: Options) {
+    super(options);
+
+    this.addEventListener(Events.applybuff.by(SELECTED_PLAYER).spell(DRAGONRAGE_TALENT), () => {
+      this.inDragonRageWindow = true;
+    });
+
+    this.addEventListener(Events.removebuff.by(SELECTED_PLAYER).spell(DRAGONRAGE_TALENT), () => {
+      this.inDragonRageWindow = false;
+    });
+
+    this.addEventListener(
+      Events.damage.by(SELECTED_PLAYER).spell([DISINTEGRATE].map((spell) => ({ id: spell.id }))),
+      () => {
+        this.totalTicks += 1;
+        if (this.inDragonRageWindow) {
+          this.dragonRageTicks += 1;
+        }
+      },
+    );
+
+    this.addEventListener(
+      Events.cast.by(SELECTED_PLAYER).spell([DISINTEGRATE].map((spell) => ({ id: spell.id }))),
+      () => {
+        this.totalCasts += 1;
+        if (this.inDragonRageWindow) {
+          this.dragonRageCasts += 1;
+        }
+      },
+    );
```

Comment:
Doing this with lambdas is definitely novel, but I guess there's nothing inherently wrong with it. I'm not sure what that `map` is about when setting the listeners tho.

For readability I think I'd prefer you use method references like all the other analyzers do though.

Reply:
The map was from a copy/paste 💦 , thanks for flagging